haystack/pipeline.py

```diff
@@ -291,6 +291,40 @@ def _overwrite_with_env_variables(cls, definition: dict):
                 param_name = key.replace(env_prefix, "").lower()
                 definition["params"][param_name] = value
 
+    def export_yaml_config(self, path: Path):
+        """
+        Save a YAML configuration for the Pipeline.
+
+        :param path: path of the output YAML file.
+        """
+        nodes = self.graph.nodes
+
+        pipeline_name = self.pipeline_type.lower()
+        pipeline_type = self.pipeline_type
+        pipelines: dict = {pipeline_name: {"name": pipeline_name, "type": pipeline_type, "nodes": []}}
+
+        components = {}
+        for node in nodes:
+            if node == self.root_node_id:
+                continue
+            component_instance = self.graph.nodes.get(node)["component"]
+            component_type = component_instance.pipeline_config["type"]
+            component_params = component_instance.pipeline_config["params"]
+            components[node] = {"name": node, "type": component_type, "params": {}}
+            for key, value in component_params.items():
+                if isinstance(value, dict) and "type" in value.keys():  # the parameter is a Component
```

Comment:
Can you add a more verbose comment here please? As I understand it, this is for the "second level" component. Not sure why it must be already a dict with "type" in it in these cases. Is it certain that the outer for loop has already come across it at this point?

Reply:
The `type` key is required to distinguish between other parameters that may be dicts. I'll add that in the docstrings.